rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -77,6 +77,28 @@ object SystemProcesses {
       }
   }
 
+  def initRegistry(
+      space: RhoISpace,
+      dispatcher: Dispatch[Task, ListChannelWithRandom, TaggedContinuation]
+  ): Seq[ListChannelWithRandom] => Task[Unit] = {
+    case Seq(ListChannelWithRandom(_, rand, cost)) =>
+      Task.now {
+        val rootPar: Par         = Registry.registryRoot
+        val rootChannel: Channel = Channel(Quote(rootPar))
+        val rootData             = space.getData(rootChannel)
```

Comment:
This won't work. Because of the way we do locking, someone may be able to race a lookup and insert a second root. It may be better to have the genesis code specifically do a produce.